code-scanning/trivy-iac-scan.yml

```
@@ -24,25 +24,23 @@ jobs:
       security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
       actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
     name: Build
-    runs-on: ubuntu-latest
+    runs-on: "ubuntu-24.04"
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Build an image from Dockerfile
-        run: |
-          docker build -t docker.io/my-organization/my-app:${{ github.sha }} .
-
-      - name: Run Trivy vulnerability scanner
-        uses: aquasecurity/trivy-action@7b7aa264d83dc58691451798b4d117d53d21edfe
+      - name: Run Trivy vulnerability scanner in IaC mode
+        uses: aquasecurity/trivy-action@6c175e9c4083a92bbca2f9724c8a5e33bc2d97a5
         with:
-          image-ref: 'docker.io/my-organization/my-app:${{ github.sha }}'
-          format: 'template'
-          template: '@/contrib/sarif.tpl'
+          scan-type: 'config'
+          hide-progress: false
+          format: 'sarif'
           output: 'trivy-results.sarif'
+          exit-code: '1'
           severity: 'CRITICAL,HIGH'
 
       - name: Upload Trivy scan results to GitHub Security tab
         uses: github/codeql-action/upload-sarif@v3
+        if: always() 
```

If you use `github/codeql-action/upload-sarif` (which this workflow does) and your workflow yields a failure (which the `exit-code: 1` would if I'm reading the docs correctly), then github's sarif stuff will say your workflow is broken, so you don't want to report an error via the workflow, instead you want to report things via upload-sarif and tell people to use a ruleset.

I describe this in my (check-spelling v0.0.25 release notes](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.25): [Code Scanning action requires a Code Scanning Ruleset](https://docs.check-spelling.dev/Breaking-Change:-Code-Scanning-action-requires-a-Code-Scanning-Ruleset)

I think this is correct:
```suggestion
          exit-code: '0'
          severity: 'CRITICAL,HIGH'

      - name: Upload Trivy scan results to GitHub Security tab
        uses: github/codeql-action/upload-sarif@v3
```